src/revocation.rs

```diff
@@ -0,0 +1,163 @@
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::fmt::Error as FormatterError;
+use std::fmt::{Debug, Display, Formatter};
+
+use crate::{basic::BasicErrorResponseType, ErrorResponseType, StandardErrorResponse};
+use crate::{AccessToken, RefreshToken};
+
+pub trait RevocableToken {
+    fn secret(&self) -> &String;
+
+    fn token_type_hint(&self) -> &'static str;
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum StandardRevocableToken {
```

Comment:
let's make this `#[non_exhaustive]` so that the crate can add more token types in the future without breaking backward compatibility

Reply:
Ah good idea.